File: TP02_Java/Quicksort/Quicksort.java
import java.io.*;
import java.time.LocalDateTime;
import java.time.Month;
import java.util.*;
import java.util.regex.*;

class CelulaDupla {
	public Time elemento;
	public CelulaDupla ant;
	public CelulaDupla prox;

	public CelulaDupla() {
		this(new Time());
	}

	public CelulaDupla(Time elemento) {
		this.elemento = elemento;
		this.ant = this.prox = null;
	}
}


class ListaTime {
	private CelulaDupla primeiro;
	private CelulaDupla ultimo;
	int mov;

	public ListaTime() {
		primeiro = new CelulaDupla();
		ultimo = primeiro;
		mov = 0;
	}

	public void inserirInicio(Time x) {
		CelulaDupla tmp = new CelulaDupla(x);

		tmp.ant = primeiro;
		tmp.prox = primeiro.prox;
		primeiro.prox = tmp;
		if(primeiro == ultimo){
			ultimo = tmp;
		}else{
			tmp.prox.ant = tmp;
		}
		tmp = null;
	}

	public void inserirFim(Time x) {
		ultimo.prox = new CelulaDupla(x);
		ultimo.prox.ant = ultimo;
		ultimo = ultimo.prox;
	}

	public Time removerInicio() throws Exception {
		if (primeiro == ultimo) {
			throw new Exception("Erro ao remover (vazia)!");
		}

		CelulaDupla tmp = primeiro;
		primeiro = primeiro.prox;
		Time resp = primeiro.elemento;
		tmp.prox = primeiro.ant = null;
		tmp = null;
		return resp;
	}

	public Time removerFim() throws Exception {
		if (primeiro == ultimo) {
			throw new Exception("Erro ao remover (vazia)!");
		} 
		Time resp = ultimo.elemento;
		ultimo = ultimo.ant;
		ultimo.prox.ant = null;
		ultimo.prox = null;
		return resp;
	}

	public void inserir(Time x, int pos) throws Exception {

		int tamanho = tamanho();

		if(pos < 0 || pos > tamanho){
			throw new Exception("Erro ao inserir posicao (" + pos + " / tamanho = " + tamanho + ") invalida!");
		} else if (pos == 0){
			inserirInicio(x);
		} else if (pos == tamanho){
			inserirFim(x);
		} else {
			// Caminhar ate a posicao anterior a insercao
			CelulaDupla i = primeiro;
			for(int j = 0; j < pos; j++, i = i.prox);

			CelulaDupla tmp = new CelulaDupla(x);
			tmp.ant = i;
			tmp.prox = i.prox;
			tmp.ant.prox = tmp.prox.ant = tmp;
			tmp = i = null;
		}
	}

	public Time remover(int pos) throws Exception {
		Time resp;
		int tamanho = tamanho();

		if (primeiro == ultimo){
			throw new Exception("Erro ao remover (vazia)!");

		} else if(pos < 0 || pos >= tamanho){
			throw new Exception("Erro ao remover (posicao " + pos + " / " + tamanho + " invalida!");
		} else if (pos == 0){
			resp = removerInicio();
		} else if (pos == tamanho - 1){
			resp = removerFim();
		} else {
			// Caminhar ate a posicao anterior a insercao
			CelulaDupla i = primeiro.prox;
			for(int j = 0; j < pos; j++, i = i.prox);

			i.ant.prox = i.prox;
			i.prox.ant = i.ant;
			resp = i.elemento;
			i.prox = i.ant = null;
			i = null;
		}

		return resp;
	}

	public void mostrar() {
		int pos = 0;
		for (CelulaDupla i = primeiro.prox; i != null; i = i.prox, pos++) {
				System.out.print("[" + pos +"] ");
				i.elemento.imprimir();
		}
	}

	public boolean pesquisar(Time x) {
		boolean resp = false;
		for (CelulaDupla i = primeiro.prox; i != null; i = i.prox) {
			if(i.elemento == x){
				resp = true;
				i = ultimo;
			}
		}
		return resp;
	}

	public int tamanho() {
		int tamanho = 0; 
		for(CelulaDupla i = primeiro; i != ultimo; i = i.prox, tamanho++);
		return tamanho;
	}

	public void swap(CelulaDupla i, CelulaDupla j) {
		Time temp = i.elemento;
		i.elemento = j.elemento;
		j.elemento = temp;
		mov += 3;
	}

	/**
	 * Algoritmo de ordenacao Quicksort.
	 */
	public void quicksort() {
		quicksort(primeiro, ultimo);
	}

	private CelulaDupla andar(int pos){
		CelulaDupla c = primeiro;
		for(int i = 0; i < pos && c.prox != null; i++, c = c.prox);
		return c;
	}

	private void quicksort(CelulaDupla esq, CelulaDupla dir) {
		CelulaDupla i = esq, j = dir;
		System.out.println(i.elemento.apelido);
		CelulaDupla pivo = dir.ant;
		while (i.elemento.apelido.compareTo(j.elemento.apelido) < 0) {
			while (i.elemento.apelido.compareTo(pivo.elemento.apelido) < 0){
				i = i.prox;
			}

			while (j.elemento.apelido.compareTo(pivo.elemento.apelido) > 0){
				j = j.ant;
			}
				
			if (i.elemento.apelido.compareTo(j.elemento.apelido) < 0) {
				swap(i, j);
				i = i.prox;
				j = j.ant;
			}
		}
		if (esq.elemento.apelido.compareTo(j.elemento.apelido) < 0)
			quicksort(esq, j);
		if (dir.elemento.apelido.compareTo(i.elemento.apelido) > 0)
			quicksort(i, dir);
	}

}

class Time {

	String nome, apelido, estadio, tecnico, liga, nomeArquivo;
	int capacidade, fundacaoDia, fundacaoMes, fundacaoAno;
	long paginaTam;

	String html;

	public Time() {
		nome = "";
		apelido = "";
		estadio = "";
		tecnico = "";
		liga = "";
		nomeArquivo = "";
		capacidade = 0;
		fundacaoDia = 0;
		fundacaoMes = 0;
		fundacaoAno = 0;
		paginaTam = 0;
		html = "";
		// MyIO.setCharset ( "ISO-8859-1" );
		MyIO.setCharset("UTF-8");
	}

	public void setNome(String novo) {
		this.nome = novo;
	}

	public String getNome() {
		return this.nome;
	}

	public void setApelido(String novo) {
		this.apelido = novo;
	}

	public String getApelido() {
		return this.apelido;
	}

	public void setEstadio(String novo) {
		this.estadio = novo;
	}

	public String getEstadio() {
		return this.estadio;
	}

	public void setTecnico(String novo) {
		this.tecnico = novo;
	}

	public String getTecnico() {
		return this.tecnico;
	}

	public void setLiga(String novo) {
		this.liga = novo;
	}

	public String getLiga() {
		return this.liga;
	}

	public void setNomeArquivo(String novo) {
		this.nome = nomeArquivo;
	}

	public String getNomeArquivo() {
		return this.nomeArquivo;
	}

	public void setCapacidade(int novo) {
		this.capacidade = novo;
	}

	public int getCapacidade() {
		return this.capacidade;
	}

	public void setFundacaoDia(int novo) {
		this.fundacaoDia = novo;
	}

	public int getFundacaoDia() {
		return this.fundacaoDia;
	}

	public void setFundacaoMes(int novo) {
		this.fundacaoMes = novo;
	}

	public int getFundacaoMes() {
		return this.fundacaoMes;
	}

	public void setFundacaoAno(int novo) {
		this.fundacaoAno = novo;
	}

	public int getFundacaoAno() {
		return this.fundacaoAno;
	}

	public void setPaginaTam(long novo) {
		this.paginaTam = novo;
	}

	public long getPaginaTam() {
		return this.paginaTam;
	}

	public void pegarHtml(String path) throws Exception {
		nomeArquivo = path;
		Arq.openRead(path, "UTF-8");

		while (!html.contains("vcard")) {
			html = Arq.readLine();
		}

		paginaTam = Arq.length();
	}

	public void pesquisarNome() {
		if (html != null) {

			Pattern pattern = Pattern.compile("Full name.*?</th><td>(.*?)<.*?", Pattern.DOTALL);
			Matcher matcher = pattern.matcher(html);

			if (matcher.find()) {
				nome = matcher.group(1).replaceAll("\\<[^>]*>", "");
			}
		}
	}

	public void pesquisarApelido() {
		if (html != null) {

			Pattern pattern = Pattern.compile("Nickname.*?</th><td class=\"nickname\">(.*?)</td></tr>",
					Pattern.DOTALL);
			Matcher matcher = pattern.matcher(html);

			if (matcher.find()) {
				apelido = matcher.group(1).replaceAll("\\<[^>]*>", "");
			}
		}
	}

	// Pesquisa a fundacao e converte para numeros
	public void pesquisarFundacao() {
		if (html != null) {

			Pattern pattern = Pattern.compile(
					"Founded.*?</th><td>(.*?)<span class=\"noprint\">.*<span class=\"bday dtstart published updated\">(.*?)</span>",
					Pattern.DOTALL);
			Matcher matcher = pattern.matcher(html);

			if (matcher.find()) {
				if (matcher.group(1).replaceAll("\\<[^>]*>", "").replace("&#160;", " ").length() == 4) {
					fundacaoAno = Integer.parseInt(matcher.group(1).replace("&#160;", " ")
							.replaceAll("\\<[^>]*>", ""));
				} else {

					Pattern pattern2 = Pattern.compile("(.*)\\s(.*)\\s(.*)$", Pattern.DOTALL);
					Matcher matcher2 = pattern2.matcher(matcher.group(1).replace("&#160;", " ")
							.replaceAll("\\<[^>]*>", ""));

					if (matcher2.find()) {
						if (matcher2.group(1).length() <= 2)
							fundacaoDia = Integer.parseInt(matcher2.group(1));

						if (matcher2.group(1).length() > 2) {
							switch (matcher2.group(1)) {
							case "January":
								fundacaoMes = 1;
								break;

							case "February":
								fundacaoMes = 2;
								break;

							case "March":
								fundacaoMes = 3;
								break;

							case "April":
								fundacaoMes = 4;
								break;

							case "May":
								fundacaoMes = 5;
								break;

							case "June":
								fundacaoMes = 6;
								break;

							case "July":
								fundacaoMes = 7;
								break;

							case "August":
								fundacaoMes = 8;
								break;

							case "September":
								fundacaoMes = 9;
								break;

							case "October":
								fundacaoMes = 10;
								break;

							case "November":
								fundacaoMes = 11;
								break;

							case "Deceber":
								fundacaoMes = 12;
								break;
							}
						} else {
							switch (matcher2.group(2)) {
							case "January":
								fundacaoMes = 1;
								break;

							case "February":
								fundacaoMes = 2;
								break;

							case "March":
								fundacaoMes = 3;
								break;

							case "April":
								fundacaoMes = 4;
								break;

							case "May":
								fundacaoMes = 5;
								break;

							case "June":
								fundacaoMes = 6;
								break;

							case "July":
								fundacaoMes = 7;
								break;

							case "August":
								fundacaoMes = 8;
								break;

							case "September":
								fundacaoMes = 9;
								break;

							case "October":
								fundacaoMes = 10;
								break;

							case "November":
								fundacaoMes = 11;
								break;

							case "Deceber":
								fundacaoMes = 12;
								break;
							}
						}

						fundacaoAno = Integer.parseInt(matcher2.group(3));
					}
				}
			}
		}
	}

	public void pesquisarEstadio() {
		if (html != null) {
			Pattern pattern = Pattern.compile("Ground.*?</th><td class=\"label\"><a.*?>(.*?)</td></tr>",
					Pattern.DOTALL);
			Matcher matcher = pattern.matcher(html);

			if (matcher.find()) {
				estadio = matcher.group(1).replaceAll("\\<[^>]*>", "");
			}
		}
	}

	public void pesquisarCapacidade() {
		if (html != null) {
			Pattern pattern = Pattern.compile("Capacity.*?</th><td>(.*?)<.*?", Pattern.DOTALL);
			Matcher matcher = pattern.matcher(html);

			if (matcher.find()) {
				capacidade = Integer.parseInt(matcher.group(1).replace(",", "").replace(".", "")
						.replace(" ", "").replaceAll("\\(.*\\)", "")
						.replaceAll("\\<[^>]*>", ""));
			}
		}
	}

	// Pesquisa o manager, se nao encontrar pesquisa pelo coach ou head coach
	public void pesquisarTecnico() {
		if (html != null) {
			Pattern pattern = Pattern.compile("oach</th><td class=\"agent\">(.*?)</td></tr><tr>",
					Pattern.DOTALL);
			Pattern pattern2 = Pattern.compile("anager</th><td class=\"agent\">(.*?)</td></tr><tr>",
					Pattern.DOTALL);

			Matcher matcher = pattern.matcher(html);
			Matcher matcher2 = pattern2.matcher(html);

			if (matcher2.find()) {
				tecnico = matcher2.group(1).replaceAll("\\<[^>]*>", "");
			} else if (matcher.find()) {
				tecnico = matcher.group(1).replaceAll("\\<[^>]*>", "");
			}
		}
	}

	public void pesquisarLiga() {
		if (html != null) {
			Pattern pattern = Pattern.compile("League</th><td>.*?\">(.*?)</a></td></tr>", Pattern.DOTALL);
			Matcher matcher = pattern.matcher(html);

			if (matcher.find()) {
				liga = matcher.group(1);
			}
		}
	}

	// Abre o arquivo e pesquisa os atributos
	public void pesquisarAtributos(String path) {
		try {
			pegarHtml(path);
		} catch (Exception e) {
			System.out.println(e);
		}

		pesquisarNome();
		pesquisarApelido();
		pesquisarFundacao();
		pesquisarEstadio();
		pesquisarCapacidade();
		pesquisarTecnico();
		pesquisarLiga();
	}

	// Imprime na tela os atributos separados por ##
	public void imprimir() {
		MyIO.print(nome + " ## ");
		MyIO.print(apelido + " ## ");

		if (fundacaoDia < 10)
			MyIO.print("0" + fundacaoDia + "/");
		else
			MyIO.print(fundacaoDia + "/");

		if (fundacaoMes < 10)
			MyIO.print("0" + fundacaoMes + "/");
		else
			MyIO.print(fundacaoMes + "/");

		MyIO.print(fundacaoAno + " ## ");

		MyIO.print(estadio + " ## ");
		MyIO.print(capacidade + " ## ");
		MyIO.print(tecnico + " ## ");
		MyIO.print(liga + " ## ");
		MyIO.print(nomeArquivo + " ## ");
		MyIO.println(paginaTam + " ##");
	}
}

class Quicksort {

	static ListaTime lista = null;

	public static void main(String[] args) {

		String[] entrada = new String[1000];
		String[] pesquisa = new String[1000];
		int numEntrada = 0;

		// Leitura da entrada padrao
		do {
			entrada[numEntrada] = MyIO.readLine();
		} while (!entrada[numEntrada++].equals("FIM"));
		numEntrada--; // Desconsiderar ultima linha contendo a palavra FIM

		lista = new ListaTime();

		for (int i = 0; i < numEntrada; i++) {
			Time time = new Time();
			try {
				time.pesquisarAtributos(entrada[i]);
				lista.inserirFim(time);
			} catch (Exception e) {
				// MyIO.println("" + e);
			}
		}

		long comeco = new Date().getTime();
		lista.quicksort();
		long fim = new Date().getTime();

		Arq.openWriteClose("66978_quicksort.txt", "UTF-8",
				"66978\t" + (fim - comeco) / 1000.0 + "\t" + lista.mov);

		lista.mostrar();
	}
}

File: TP02_C/Quicksort/Quicksort.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

#define bool   short
#define true   1
#define false  0

typedef struct
{
	char nome[1024];
	char *apelido;
	char *estadio;
	char *tecnico;
	char liga[1024];
	char *nomeArquivo;

	int capacidade;
	int fundacaoDia;
	int fundacaoMes;
	int fundacaoAno;

	long paginaTam;

} Time;

void imprimir(Time *time)
{
	printf("%s ## %s ## ", time->nome, time->apelido);
	printf("%02d/%02d/%04d ## ", time->fundacaoDia, time->fundacaoMes, time->fundacaoAno);
	printf("%s ## %i ## ", time->estadio, time->capacidade);
	printf("%s ## %s ## ", time->tecnico, time->liga);
	printf("%s ## %li ##\n", time->nomeArquivo, time->paginaTam);
}


Time* novoTime()
{
	Time* time = (Time*)malloc(sizeof(Time));


		//time->nome = (char *)malloc(1024 * sizeof(char));
		time->apelido = (char *)malloc(1024 * sizeof(char));
		time->estadio = (char *)malloc(1024 * sizeof(char));
		time->tecnico = (char *)malloc(1024 * sizeof(char));
		//time->liga = (char *)malloc(10000 * sizeof(char));
		time->nomeArquivo = (char *)malloc(1024 * sizeof(char));

		time->capacidade = 0;
		time->fundacaoDia = 0;
		time->fundacaoMes = 0;
		time->fundacaoAno = 0;

		time->paginaTam = 0;
	

	return time;
}

//TIPO CELULA ===================================================================
typedef struct Celula {
	Time* elemento;        // Elemento inserido na celula.
	struct Celula* prox; // Aponta a celula prox.
} Celula;

Celula* novaCelula(Time* elemento) {
   Celula* nova = (Celula*) malloc(sizeof(Celula));
   nova->elemento = elemento;
   nova->prox = NULL;
   return nova;
}

//LISTA PROPRIAMENTE DITA =======================================================
Celula* primeiro;
Celula* ultimo;

void start () {
   primeiro = novaCelula(novoTime());
   ultimo = primeiro;
}

void inserirInicio(Time* x) {
   Celula* tmp = novaCelula(x);
   tmp->prox = primeiro->prox;
   primeiro->prox = tmp;
   if (primeiro == ultimo) {                    
      ultimo = tmp;
   }
   tmp = NULL;
}

void inserirFim(Time* x) {
   ultimo->prox = novaCelula(x);
   ultimo = ultimo->prox;
}

Time* removerInicio() {
   if (primeiro == ultimo) {
   }

   Celula* tmp = primeiro;
   primeiro = primeiro->prox;
   Time* resp = primeiro->elemento;
   tmp->prox = NULL;
   free(tmp);
   tmp = NULL;
   return resp;
}

Time* removerFim() {
   if (primeiro == ultimo) {
   } 

   // Caminhar ate a penultima celula:
   Celula* i;
   for(i = primeiro; i->prox != ultimo; i = i->prox);

   Time* resp = ultimo->elemento;
   ultimo = i;
   free(ultimo->prox);
   i = ultimo->prox = NULL;

   return resp;
}

int tamanho() {
   int tamanho = 0;
   Celula* i;
   for(i = primeiro; i != ultimo; i = i->prox, tamanho++);
   return tamanho;
}

void inserir(Time* x, int pos) {

   int tam = tamanho();

   if(pos < 0 || pos > tam){
   } else if (pos == 0){
      inserirInicio(x);
   } else if (pos == tam){
      inserirFim(x);
   } else {
      // Caminhar ate a posicao anterior a insercao
      int j;
      Celula* i = primeiro;
      for(j = 0; j < pos; j++, i = i->prox);

      Celula* tmp = novaCelula(x);
      tmp->prox = i->prox;
      i->prox = tmp;
      tmp = i = NULL;
   }
}

Time* remover(int pos) {
   Time* resp;
   int tam = tamanho();

   if (primeiro == ultimo){
   } else if(pos < 0 || pos >= tam){
   } else if (pos == 0){
      resp = removerInicio();
   } else if (pos == tam - 1){
      resp = removerFim();
   } else {
      // Caminhar ate a posicao anterior a insercao
      Celula* i = primeiro;
      int j;
      for(j = 0; j < pos; j++, i = i->prox);

      Celula* tmp = i->prox;
      resp = tmp->elemento;
      i->prox = tmp->prox;
      tmp->prox = NULL;
      free(tmp);
      i = tmp = NULL;
   }
   return resp;
}

void mostrar() {
   Celula* i;
   for (i = primeiro->prox; i != NULL; i = i->prox) {
      imprimir(i->elemento);
   }
}

bool pesquisar(Time* x) {
   bool retorno = false;
   Celula* i;

   for (i = primeiro->prox; i != NULL; i = i->prox) {
      if(i->elemento == x){
         retorno = true;
         i = ultimo;
      }
   }
   return retorno;
}


void swap(Celula* i, Celula* j) {
   Time* temp = i->elemento;
   i->elemento = j->elemento;
  j->elemento = temp;
}

Celula* andar(int x){
	Celula* c = primeiro;
	for(int i = 0; i < x && c != NULL; i++, c = c->prox);
	return c;
}

void quicksortRec (int esq, int dir) {
   Time* i = andar(esq)->elemento;
   Time* j = andar(dir)->elemento;
   Time* pivo = andar((dir+esq)/2)->elemento;
   while (strcmp(i->apelido, j->apelido) <= 0) {
      while (array[i] < pivo) i++;
      while (array[j] > pivo) j--;
      if (i <= j) {
         swap(i, j);
         i++;
         j--;
      }
   }
   if (esq < j)  quicksortRec(esq, j);
   if (i < dir)  quicksortRec(i, dir);
}

// Algoritmo de ordenacao
void quicksort() {
   quicksortRec(0, tamanho());
}

char *substring(char *padrao, char *entrada)
{
	char *pointer = strstr(entrada, padrao);

	return strdup(pointer);
}

long int indexOf(char *padrao, char *entrada)
{
	char *pointer = strstr(entrada, padrao);

	return pointer - entrada;
}

bool contains(char *string, char *procura)
{
	return strstr(string, procura) != NULL;
}

char *removerTags(char *str)
{
	char *output = (char *)malloc(10000 * sizeof(char));
	int current = 0;

	bool isTag = false;
	bool isSpecial = false;

	for (int i = 0; i < strlen(str); i++)
	{
		if (str[i] == '<')
			isTag = 1;
		else if (str[i] == '>')
		{
			isTag = 0;

			if (i > 0 && current > 0 && output[current-1] != '@' && str[i-1] != 'i' && str[i-1] != '/' && output[current-1])
			{
				output[current] = '@';
				current++;
			}
		}
		else if (isTag == 0 && str[i] == '&')
			isSpecial = 1;
		else if (isTag == 0 && str[i] == ';')
			isSpecial = 0;
		else if (isTag == 0 && isSpecial == 0)
		{
			output[current] = str[i];
			current++;
		}
	}

	return strdup(output);
}

char *searchUp(char *entrada, char *padrao, char *final)
{
	char tmp[1024] = "";
	char *start, *end;

	strcpy(tmp, "");

	if (start = strstr(entrada, padrao))
	{
		start += strlen(padrao);
		if (end = strstr(start, final))
		{
			//tmp = (char *)malloc(end - start + 1);
			memcpy(tmp, start, end - start);
			tmp[end - start] = '\0';
		}
	}

	return strdup(tmp);
	//free(start);
	//free(end);
}

void pegarCapacidade(Time *time, char *linha)
{
	char *tmp = (char *)malloc(1000 * sizeof(char));
	char *numero = (char *)malloc(256 * sizeof(char));

	strcpy(tmp, searchUp(linha, "Capacity@", "@"));

	int j = 0;
	for (int i = 0; i < strlen(tmp); i++)
	{
		if (tmp[i] != ',' && tmp[i] != ' ')
		{
			numero[j] = tmp[i];
			j++;
		}
	}

	sscanf(numero, "%d", &(time->capacidade));	
	//free(tmp);
}

void pegarData(Time *time, char *linha)
{
	char *tmp = (char *)malloc(1000 * sizeof(char));
	strcpy(tmp, searchUp(linha, "Founded@", "@"));

	if (strlen(tmp) == 4)
        	sscanf(tmp,"%d",&(time->fundacaoAno));
	else{
		if(contains(tmp, "anuary")){
			time->fundacaoMes = 1;
			sscanf(searchUp(tmp, "anuary", "\0"),"%d",&(time->fundacaoAno));
		}else if(contains(tmp, "ebruary")){
			time->fundacaoMes = 2;
			sscanf(searchUp(tmp, "ebruary", "\0"),"%d",&(time->fundacaoAno));
		}else if(contains(tmp, "arch")){
			time->fundacaoMes = 3;
			sscanf(searchUp(tmp, "arch", "\0"),"%d",&(time->fundacaoAno));
		}else if(contains(tmp, "pril")){
			time->fundacaoMes = 4;	
			sscanf(searchUp(tmp, "pril", "\0"),"%d",&(time->fundacaoAno));		
		}else if(contains(tmp, "ay")){
			time->fundacaoMes = 5;	
			sscanf(searchUp(tmp, "ay", "\0"),"%d",&(time->fundacaoAno));	
		}else if(contains(tmp, "une")){
			time->fundacaoMes = 6;	
			sscanf(searchUp(tmp, "une", "\0"),"%d",&(time->fundacaoAno));		
		}else if(contains(tmp, "uly")){
			time->fundacaoMes = 7;	
			sscanf(searchUp(tmp, "uly", "\0"),"%d",&(time->fundacaoAno));	
		}else if(contains(tmp, "ugust")){
			time->fundacaoMes = 8;	
			sscanf(searchUp(tmp, "ugust", "\0"),"%d",&(time->fundacaoAno));	
		}else if(contains(tmp, "eptember")){
			time->fundacaoMes = 9;	
			sscanf(searchUp(tmp, "eptember", "\0"),"%d",&(time->fundacaoAno));
		}else if(contains(tmp, "ctober")){
			time->fundacaoMes = 10;	
			sscanf(searchUp(tmp, "ctober", "\0"),"%d",&(time->fundacaoAno));
		}else if(contains(tmp, "ovember")){
			time->fundacaoMes = 11;	
			sscanf(searchUp(tmp, "ovember", "\0"),"%d",&(time->fundacaoAno));
		}else if(contains(tmp, "ecember")){
			time->fundacaoMes = 12;	
			sscanf(searchUp(tmp, "ecember", "\0"),"%d",&(time->fundacaoAno));
		}
	}
	
	//free(tmp);
}

int lerHtml(char *arquivo, Time *time)
{
	strcpy(time->nomeArquivo, arquivo);
	FILE *file = fopen(arquivo, "r");

	if (file)
	{
		char *init = (char *)malloc(128 * sizeof(char));
		char *line = (char *)malloc(10000 * sizeof(char));

		fgets(init, 49, file);

		while (!contains(init, "<table class=\"infobox vcard\" style=\"width:22em\">"))
		{
			if (strcmp("</html>\n", init) == 0)
				return 1;

			if (contains(init, "put\"><table class=\"infobox vcard\" style=\"width:2"))
				break;

			fgets(init, 49, file);
		}

		fgets(line, 10000, file);

		strcpy(line, removerTags(line));

		//printf("%s\n", line);

		//printf("%s\n", searchUp(line, "Full name@", "@"));
		strcpy(time->nome, searchUp(line, "Full name@", "@"));
		//printf("%s\n", time->nome);
		strcpy(time->apelido, searchUp(line, "Nickname(s)@", "@"));
		//printf("%s\n", time->apelido);
		strcpy(time->estadio, searchUp(line, "Ground@", "@"));
		//printf("%s\n", time->estadio);

		strcpy(time->tecnico, searchUp(line, "oach@", "@"));

		if (strcmp(time->tecnico, "") == 0)
		{
			strcpy(time->tecnico, searchUp(line, "anager@", "@"));
		}

		//printf("%s\n", time->tecnico);

		strcpy(time->liga, searchUp(line, "League@", "@"));
		//printf("%s\n", time->liga);

		pegarCapacidade(time, line);
		pegarData(time, line);
		//printf("%i", time->capacidade);

		fseek(file, 0, SEEK_END);
		time->paginaTam = ftell(file);
		fclose(file);
	}

	return 0;
}

int main()
{
	start();
	char *arquivo = (char *)malloc(256 * sizeof(char));
	char *arquivo2 = (char *)malloc(512 * sizeof(char));

	scanf("%s", arquivo);
	
	while (arquivo != NULL && strcmp(arquivo, "FIM") != 0)
	{
		Time* t = novoTime();
		lerHtml(arquivo, t);
		inserirFim(t);
		scanf("%s", arquivo);
	}

	mostrar();

	//getchar();
	//getchar();

	return 0;
}